instchown.c

```diff
@@ -30,46 +31,65 @@ static void ddhome(stralloc *dd, const char *home)
 
 void h(char *home, uid_t uid, gid_t gid, int mode)
 {
+  int fd;
   stralloc dh = { 0 };
   ddhome(&dh, home);
   home=dh.s;
-  if (chown(home,uid,gid) == -1)
-    strerr_die4sys(111,FATAL,"unable to chown ",home,": ");
-  if (chmod(home,mode) == -1)
-    strerr_die4sys(111,FATAL,"unable to chmod ",home,": ");
+  if ((fd = open_read(home))) {
```

Comment:
This will not work at all because on error this returns -1 which is "true". You need ```((fd = open_read(home)) >= 0)```.